harvester/GeoNetworkRepository.py

```diff
@@ -0,0 +1,329 @@
+from harvester.HarvestRepository import HarvestRepository
+from harvester.rate_limited import rate_limited
+import time
+import requests
+import lxml.etree as ET
+from rdflib import Graph, DCAT
+from dateutil import parser
+import urllib
+import re
+
+class GeoNetworkRepository(HarvestRepository):
+    """ GeoNetwork Repository """
+
+    def setRepoParams(self, repoParams):
+        self.metadataprefix = "geonetwork"
+        super(GeoNetworkRepository, self).setRepoParams(repoParams)
+        self.domain_metadata = []
+        self.headers = {'accept': 'application/rdf+xml'}
+
+    def _crawl(self):
+        kwargs = {
+            "repo_id": self.repository_id, "repo_url": self.url, "repo_set": self.set, "repo_name": self.name,
+            "repo_type": "geonetwork",
+            "enabled": self.enabled, "repo_thumbnail": self.thumbnail, "item_url_pattern": self.item_url_pattern,
+            "abort_after_numerrors": self.abort_after_numerrors,
+            "max_records_updated_per_run": self.max_records_updated_per_run,
+            "update_log_after_numitems": self.update_log_after_numitems,
+            "record_refresh_days": self.record_refresh_days,
+            "repo_refresh_days": self.repo_refresh_days, "homepage_url": self.homepage_url,
+            "repo_oai_name": self.repo_oai_name,
+            "repo_registry_uri": self.repo_registry_uri
+        }
+        self.repository_id = self.db.update_repo(**kwargs)
+
+        try:
+            g = Graph()
+            request_success = False
+            request_count = 0
+            while not request_success and request_count < 5:
+                try:
+                    g.parse(self.url)
+                    request_success = True
+                except urllib.error.HTTPError as e:
```

Comment:
@prairiewest is this a decent way to handle this, given reliability issues?

Reply:
Sure. You may want to sleep for longer than 1 second - maybe 3?